tests/integration/modules/test_win_lgpo.py

```diff
@@ -506,6 +506,22 @@ def test_set_computer_policy_ClipboardRedirection(self):
             'Not Configured',
             [r'; Source file:  c:\\windows\\system32\\grouppolicy\\machine\\registry.pol[\s]*; PARSING COMPLETED.'])
 
+    @destructiveTest
+    def test_set_computer_policy_GuestAccountStatus(self):
+        '''
+        Test setting/unsetting/changing GuestAccountStatus
+        '''
+        # disable GuestAccountStatus
+        self._testSeceditPolicy(
+            'GuestAccountStatus',
+            'Disabled',
+            [r'^EnableGuestAccount = 0'])
```

Comment:
Would this function exhibit the error if we called it a second time? Or should it be failing currently?

We definitely should have a test that shows the KeyError when the change in 5529 is **not** present.

Reply:
I don't understand what you're saying. This test *will* throw the KeyError if the change in `win_lgpo.py` is *not* present. 5529 is a meaningless number to me, without some kind of reference link.

Myself, I've worked around this differently by using my own custom execution module. If you want to write another test or change the code, I've ticked the box to allow edits from maintainers. Code is easier to understand than words.

Reply:
To clarify:

if someone changes the code back to:

      __context__.pop('lgpo.secedit_data')

Or something equivalent, we need a regression test to show that.

Reply:
Sorry, 5529 was the line of code that you changed 🙃

Reply:
Yes, that is _exactly_ what this test does.

Reply:
Sorry, I was trying to get at that in my prior comment:

> Just added a test. Turns out this wasn't caught before because the tests were only exercising the `_modal_sets` logic path in `set_`... This bug is in the `_secedits` logic path.

There were tests for this module, and even this function, but the tests were not exercising this logic path in `set_`. The test I've added _does_ exercise the logic path, so without this change in `win_lgpo.py` the test will fail.

Reply:
Gotcha - It wasn't obvious that `self._testSeceditPolicy(` was in the `_secedits` logic path, from what was shown in the PR here on GitHub. I *guessed* that your test would expose it, but without checking out the PR and futzing with the tests, or just asking you directly, I wasn't sure.

Now we're both on the same page :) Thanks for responding!

Reply:
I get it. It took me a while to trace the code and figure out how the tests worked and why this wasn't caught before. The `set_` function could use some serious refactoring to separate these concerns and focus on doing one thing...

Reply:
There are definitely a number of places that could use similar refactoring. If you're interested in doing any of that refactoring for this section of the code, we've got a couple of months before the Sodium release - if you're available during one of the Community Open Hours to discuss what you're thinking. If you're not available during that period, I think just opening up an issue documenting the changes you'd like to make would be good enough.